src/components/dashboard/Dashboard.js

```diff
@@ -116,19 +126,54 @@ export class Dashboard extends Component {
     );
   };
 
+  renderUserFilter = () => {
+    const { database, t, userMode } = this.props;
+    const { filteredUserId } = this.state;
+
+    if (!database || _.isEmpty(database)) {
```

Comment:
Can this not lead to infinite loading for empty database? I think we had a solution for this elsewhere.